### ISSUE TITLE
UnicodeDecodeError on Windows

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#2994]
### Observed behavior

Server fails to start and open in the browser after installation on Windows 7/10. 
Reported on our [Community Forums](https://community.learningequality.org/t/kolibri-the-server-is-starting-please-wait-kolibri-exe/412).

### Errors and logs

Output when started from the command line:

```
C:\Python27\Scripts>kolibri start --debug --foreground
e[37mINFO Running Kolibri with the following settings: kolibri.deployment.default.settings.basee[0m
e[37mINFO Running update routines for new version…e[0m

0 static files copied to ‘C:\Users\enock.kolibri\static’, 407 unmodified.
Operations to perform:
Apply all migrations: sessions, admin, kolibriauth, morango, auth, content, contenttypes, exams, device, logger
Running migrations:
No migrations to apply.
Your models have changes that are not yet reflected in a migration, and so won’t be applied.
Run ‘manage.py makemigrations’ to make new migrations, and then re-run ‘manage.py migrate’ to apply them.
e[37mINFO Running ‘kolibri start’ in foreground…e[0m
Traceback (most recent call last):
File “c:\python27\lib\runpy.py”, line 174, in _run_module_as_main
"main", fname, loader, pkg_name)
File “c:\python27\lib\runpy.py”, line 72, in run_code
exec code in run_globals
File "C:\Python27\Scripts\kolibri.exe_main.py", line 9, in
File “c:\python27\lib\site-packages\kolibri\utils\cli.py”, line 655, in main
start(port, daemon=daemon)
File “c:\python27\lib\site-packages\kolibri\utils\cli.py”, line 326, in start
_, urls = server.get_urls(listen_port=port)
File “c:\python27\lib\site-packages\kolibri\utils\server.py”, line 297, in get_urls
interfaces = ifcfg.interfaces()
File "c:\python27\lib\site-packages\kolibri\dist\ifcfg_init.py", line 68, in interfaces
return Parser(ifconfig=ifconfig).interfaces
File “c:\python27\lib\site-packages\kolibri\dist\ifcfg\parser.py”, line 29, in init
self.parse(self.ifconfig_data)
File “c:\python27\lib\site-packages\kolibri\dist\ifcfg\parser.py”, line 53, in parse
ifconfig, __, __ = exec_cmd(self.get_command())
File “c:\python27\lib\site-packages\kolibri\dist\ifcfg\tools.py”, line 32, in exec_cmd
stdout = stdout.decode(locale.getpreferredencoding()).encode(‘utf-8’)
File “c:\python27\lib\encodings\cp1252.py”, line 15, in decode
return codecs.charmap_decode(input,errors,decoding_table)
UnicodeDecodeError: ‘charmap’ codec can’t decode byte 0x81 in position 856: character maps to

C:\Python27\Scripts>
```

### Steps to reproduce

I was not able to reproduce this error while testing on virtual machines.

### Context

 * Kolibri version: 0.7.0
 * Operating system: Windows 7/10

cc @mrpau-richard @benjaoming @rtibbles

**Comment by benjaoming** [comment]
This should be reported to the idiot maintaining `ifcfg`... oh.... https://github.com/learningequality/python-ifcfg

The fix should be manageable, will have a look and create a new release!

**Comment by radinamatic** [comment]
Once I managed to install the German language pack on Windows 10, the error is replicable, exactly as the OP reported:

1. Installed the [dev build](https://storage.googleapis.com/le-downloads/kolibri%2Frelease%2F4607%2Freport.html) by @benjaoming 
2. Notification `Kolibri is starting, please wait...` keeps popping out , but server is not starting and the browser is not opening.
3. Stopped and tried to start from the command prompt, got the original `UnicodeDecodeError`
4. Deleted the `./kolibri` folder and tried again, same arror.

![win10 start running - oracle vm virtualbox_136](https://user-images.githubusercontent.com/1457929/34314582-558ba3e4-e775-11e7-9687-cad0da811d08.png)

**Comment by benjaoming** [comment]
@radinamatic while fixing the encoding issue on Windows, I would love to also ensure that there isn't any other translations "gotchas" -- would you be able to PM me the output of the command `ipconfig` as raw copied text from the terminal?

**Comment by radinamatic** [comment]
@benjaoming 

VM networks by default:

![win10 0 7 ui de running - oracle vm virtualbox_147](https://user-images.githubusercontent.com/1457929/34353971-6d667efe-ea2b-11e7-8f20-7286a2cc01ce.png)

After adding one bridged adapter so the VM gets assigned an IP from my LAN router:

![win10 0 7 ui de running - oracle vm virtualbox_148](https://user-images.githubusercontent.com/1457929/34353994-983e0566-ea2b-11e7-86f1-2cfa2a0937b0.png)

```Microsoft Windows [Version 10.0.16299.125]
(c) 2017 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\IEUser>ipconfig

Windows-IP-Konfiguration


Ethernet-Adapter Ethernet:

   Verbindungsspezifisches DNS-Suffix:
   Verbindungslokale IPv6-Adresse  . : fe80::c995:fc34:c5c2:933a%2
   IPv4-Adresse  . . . . . . . . . . : 10.0.2.15
   Subnetzmaske  . . . . . . . . . . : 255.255.255.0
   Standardgateway . . . . . . . . . : 10.0.2.2

Ethernet-Adapter Ethernet 2:

   Verbindungsspezifisches DNS-Suffix:
   Verbindungslokale IPv6-Adresse  . : fe80::c866:ba08:6d5a:f8db%16
   IPv4-Adresse  . . . . . . . . . . : 192.168.1.43
   Subnetzmaske  . . . . . . . . . . : 255.255.255.0
   Standardgateway . . . . . . . . . : 192.168.1.1

Tunneladapter Teredo Tunneling Pseudo-Interface:

   Verbindungsspezifisches DNS-Suffix:
   IPv6-Adresse. . . . . . . . . . . : 2001:0:9d38:6abd:30c2:2ddd:3f57:fed4
   Verbindungslokale IPv6-Adresse  . : fe80::30c2:2ddd:3f57:fed4%12
   Standardgateway . . . . . . . . . : ::

C:\Users\IEUser>```

**Comment by benjaoming** [comment]
Oh no, the error does not reproduce from the german output :/

But there are some other issues possibly with these translations, which I have to look at, basically the parsing assumes the output is in English.

Tracking both here:
https://github.com/ftao/python-ifcfg/issues/17
https://github.com/ftao/python-ifcfg/issues/18

**Comment by indirectlylit** [comment]
merged #3002

should we close this and create new issues for the i18n ifconfig, or is that reasonably part of this issue?

**Comment by benjaoming** [comment]
@indirectlylit since nature of the error in this issue persists, there's no need to open a new issue

It's really crazy that we're in 2018, and I'm spending 2 days on fixing encoding issues.

**Comment by benjaoming** [comment]
Fixed in #3070 and to be released in 0.7.1
